velox/functions/sparksql/specialforms/FromJson.cpp

```diff
@@ -0,0 +1,591 @@
+                  rowType.childAt(index)->kind(),
+                  field.value(),
+                  writerTyped.get_writer_at(index),
+                  false);
+              if (res != simdjson::SUCCESS) {
+                writerTyped.set_null_at(index);
+              }
+            }
+          }
+        }
+
+        for (const auto& [key, index] : fieldIndices) {
+          if (index >= 0) {
+            writerTyped.set_null_at(index);
+          }
+        }
+      } else {
+        if (isRoot) {
+          writerTyped.set_null_at(0);
+          return simdjson::SUCCESS;
+        } else {
+          return simdjson::INCORRECT_TYPE;
+        }
+      }
+      return simdjson::SUCCESS;
+    }
+  };
+
+  static simdjson::simdjson_result<std::string_view> rawJson(
+      Input value,
+      simdjson::ondemand::json_type type) {
+    switch (type) {
+      case simdjson::ondemand::json_type::array: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto array, value.get_array());
+        return array.raw_json();
+      }
+      case simdjson::ondemand::json_type::object: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto object, value.get_object());
+        return object.raw_json();
+      }
+      default:
+        return value.raw_json_token();
+    }
+  }
+
+  template <typename T>
+  static simdjson::error_code castJsonToInt(
+      Input value,
+      exec::GenericWriter& writer) {
+    SIMDJSON_ASSIGN_OR_RAISE(auto type, value.type());
+    switch (type) {
+      case simdjson::ondemand::json_type::number: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto num, value.get_number());
+        switch (num.get_number_type()) {
+          case simdjson::ondemand::number_type::floating_point_number:
+            return simdjson::INCORRECT_TYPE;
+          case simdjson::ondemand::number_type::signed_integer:
+            return convertIfInRange<T>(num.get_int64(), writer);
+          case simdjson::ondemand::number_type::unsigned_integer:
+            return simdjson::NUMBER_OUT_OF_RANGE;
+          case simdjson::ondemand::number_type::big_integer:
+            VELOX_UNREACHABLE(); // value.get_number() would have failed
+                                 // already.
+        }
+        break;
+      }
+      default:
+        return simdjson::INCORRECT_TYPE;
+    }
+    return simdjson::SUCCESS;
+  }
+
+  template <typename T>
+  static simdjson::error_code castJsonToFloatingPoint(
+      Input value,
+      exec::GenericWriter& writer) {
+    SIMDJSON_ASSIGN_OR_RAISE(auto type, value.type());
+    switch (type) {
+      case simdjson::ondemand::json_type::number: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto num, value.get_double());
+        return convertIfInRange<T>(num, writer);
+      }
+      case simdjson::ondemand::json_type::string: {
+        SIMDJSON_ASSIGN_OR_RAISE(auto s, value.get_string());
+        constexpr T kNaN = std::numeric_limits<T>::quiet_NaN();
+        constexpr T kInf = std::numeric_limits<T>::infinity();
+        if (s == "NaN") {
```

Comment:
Do we need a case-insensitive comparison?

Reply:
Here the comparison is case sensitive.
![image](https://github.com/user-attachments/assets/a977aa9e-ee1e-4b32-9385-d1e7dbd19fa7)